README.md

```diff
@@ -36,3 +36,157 @@ This is the shared library defining public classes that constitute the worker in
 
 This contains implementations of the testing framework to allow for integration testing of the Document Worker. The project can be found in [worker-document-testing](worker-document-testing).
+
+## Workflow event handlers
+
+The below are the various event handlers from the workflow scripts.
+
+It is not mandatory that the workflow script file should contain all of the below event handlers.
+Sometimes the workflow script may contain only few handlers from the below list.
+The event handler will only be triggered, if there is corresponding function from the JavaScript file being executed.
+
+####  onProcessTask
+
+This is the first function called by worker on the task message.
+This function is passed `TaskEventObject` with `Task` as an argument.
+
+An example of the `TaskEventObject` is below. The variables will be initialized with values from task message and it will be sent to the series of other functions in the workflow for further processing.
+```
+TaskEventObject(Task task){
+    application=task.application;
+    task = task;
+    rootDocument=task.document;
+}
+```
+For more details of the <a name="onProcessTask"></a>`TaskEventObject` refer the java implementation of the class, [TaskEventObject.java](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/TaskEventObject.java) 
+
+```
+/* global thisScript */
+
+function onProcessTask(e)
+{	
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+}
+```
+
+#### onBeforeProcessDocument
+
+This event will be executed after `onProcessTask` and before processing of a document. 
+
+This function is passed `CancelableDocumentEventObject`  with `Document` as an argument.
+
+An example of the `CancelableDocumentEventObject` is below.
+```
+CancelableDocumentEventObject(Document document)
+{
+    super(document);
+    cancel=false;
+}
+```
+For more details of the `CancelableDocumentEventObject`,  refer the Java implementation of for the class,
+[CancelableDocumentEventObject.java](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/CancelableDocumentEventObject.java) 
+
+```
+function onBeforeProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+    // e.cancel       (writable)  (default: false)
+}
+```
+Output of this function would be the boolean value of event's cancel flag.
+This flag is used to determine if that individual document should be processed by the worker.
+
+#### <a name="onProcessDocument"></a>onProcessDocument
+
+This function will be called to process a document. 
+This function is passed `DocumentEventObject` with `Document` as an argument.
+The `onBeforeProcessDocument`  event has already been triggered to check for cancellation flag.
```

Comment:
"The `onBeforeProcessDocument`  event has already been triggered to check for cancellation flag."

Extra space in this line and is it really necessary?